src/main/java/seedu/address/model/person/Role.java

```diff
@@ -49,6 +51,24 @@ public static boolean isValidRole(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns person with its particular type based on the role.
+     *
+     * @param name Name of person.
+     * @param phone Phone of person.
+     * @param email Email of person.
+     * @param affiliationList Affiliations of person.
+     */
+    public Person generatePerson(Name name, Phone phone, Email email, Set<Affiliation> affiliationList) {
+        if (value.toUpperCase().equals(Type.DOCTOR.name())) {
+            return new Doctor(name, phone, email, affiliationList);
+        } else if (value.toUpperCase().equals(Type.PATIENT.name())) {
+            return new Patient(name, phone, email, affiliationList);
+        } else {
+            return null;
+        }
+    }
```

Comment:
As we proceed to use different Person classes, it would be ideal to have the Role class itself generate the correct Person class type. This is in comparison to iterating through the different Person roles and then initialise the correct class during parsing.

However, as the role may intuitively be an attribute of the person, having the Role class generate the Person may not be logical. Open to other possible implementations!

Reply:
Yup, I think that is helpful!